containers-tests/tests/map-strictness.hs

```diff
@@ -82,6 +94,35 @@ pInsertLookupWithKeyValueStrict f k v m
                      not (isBottom $ M.insertLookupWithKey (const3 1) k bottom m)
     | otherwise    = isBottom $ M.insertLookupWithKey (apply3 f) k bottom m
 
+#if __GLASGOW_HASKELL__ >= 806
+pStrictFoldr' :: Map Int Int -> Property
+pStrictFoldr' m = ioProperty $ isNothing <$> noThunks [] r
+  where
+    !r = M.foldr' (:) [] m
+#endif
+
+
+#if __GLASGOW_HASKELL__ >= 806
+pStrictFoldl' :: Map Int Int -> Property
+pStrictFoldl' m = ioProperty $ isNothing <$> noThunks [] r
+  where
+    !r = M.foldl' (flip (:)) [] m
+#endif
+
+#if __GLASGOW_HASKELL__ >= 806
+pStrictFoldrWithKey' :: Map Int Int -> Property
+pStrictFoldrWithKey' m = ioProperty $ isNothing <$> noThunks [] r
+  where
+    !r = M.foldrWithKey' (\_ a as -> a : as) [] m
+#endif
+
+#if __GLASGOW_HASKELL__ >= 806
+pStrictFoldlWithKey' :: Map Int Int -> Property
+pStrictFoldlWithKey' m = ioProperty $ isNothing <$> noThunks [] r
+  where
+    !r = M.foldlWithKey' (\as _ a -> a : as) [] m
+#endif
```

Comment:
Is there an advantage to having one CPP conditional per property? It seems to me that a single one for all of them would be sufficient.

Reply:
I prefer short `#if`'s over long ones that are easier to miss when reading the code.

Reply:
Good point.